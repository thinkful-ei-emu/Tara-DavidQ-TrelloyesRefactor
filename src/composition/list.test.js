import React from 'react';
import ReactDOM from 'react-dom';
import List from './list.js';
import renderer from 'react-test-renderer';

describe('List Component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<List key='a' cardIds= {['a', 'b', 'c']} header={'List 1'} allCards={{
            'a': { id: 'a', title: 'First card', content: 'lorem ipsum' },
            'b': { id: 'b', title: 'Second card', content: 'lorem ipsum' },
            'c': { id: 'c', title: 'Third card', content: 'lorem ipsum' },
            'd': { id: 'd', title: 'Fourth card', content: 'lorem ipsum' },
            'e': { id: 'e', title: 'Fifth card', content: 'lorem ipsum' },
            'f': { id: 'f', title: 'Sixth card', content: 'lorem ipsum' },
            'g': { id: 'g', title: 'Seventh card', content: 'lorem ipsum' },
            'h': { id: 'h', title: 'Eighth card', content: 'lorem ipsum' },
            'i': { id: 'i', title: 'Ninth card', content: 'lorem ipsum' },
            'j': { id: 'j', title: 'Tenth card', content: 'lorem ipsum' },
            'k': { id: 'k', title: 'Eleventh card', content: 'lorem ipsum' },
            'l': { id: 'l', title: 'Twelfth card', content: 'lorem ipsum' },
            'm': { id: 'm', title: 'Thirteenth card', content: 'lorem ipsum' },
          }} />, div);

          ReactDOM.unmountComponentAtNode(div);
    })

    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<List key='a' cardIds= {['a', 'b', 'c']} header={'List 1'} allCards={{
                'a': { id: 'a', title: 'First card', content: 'lorem ipsum' },
                'b': { id: 'b', title: 'Second card', content: 'lorem ipsum' },
                'c': { id: 'c', title: 'Third card', content: 'lorem ipsum' },
                'd': { id: 'd', title: 'Fourth card', content: 'lorem ipsum' },
                'e': { id: 'e', title: 'Fifth card', content: 'lorem ipsum' },
                'f': { id: 'f', title: 'Sixth card', content: 'lorem ipsum' },
                'g': { id: 'g', title: 'Seventh card', content: 'lorem ipsum' },
                'h': { id: 'h', title: 'Eighth card', content: 'lorem ipsum' },
                'i': { id: 'i', title: 'Ninth card', content: 'lorem ipsum' },
                'j': { id: 'j', title: 'Tenth card', content: 'lorem ipsum' },
                'k': { id: 'k', title: 'Eleventh card', content: 'lorem ipsum' },
                'l': { id: 'l', title: 'Twelfth card', content: 'lorem ipsum' },
                'm': { id: 'm', title: 'Thirteenth card', content: 'lorem ipsum' },
              }} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
})