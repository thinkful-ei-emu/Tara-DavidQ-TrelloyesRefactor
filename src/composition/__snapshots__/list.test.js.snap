// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Component renders the UI as expected 1`] = `
<div
  className="List"
>
  <header
    className="List-header"
  >
    List 1
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
      id="a"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        First card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
      id="b"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        Second card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
      id="c"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        Third card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
  </div>
</div>
`;
